app/celery/provider_tasks.py

```diff
@@ -70,3 +89,68 @@ def deliver_email(self, notification_id):
             )
             update_notification_status_by_id(notification_id, NOTIFICATION_TECHNICAL_FAILURE)
             raise NotificationTechnicalFailureException(message)
+
+
+@notify_celery.task(bind=True, name="deliver_letter", max_retries=55, retry_backoff=True, retry_backoff_max=300)
+def deliver_letter(self, notification_id):
+    # 55 retries with exponential backoff gives a retry time of approximately 4 hours
+    current_app.logger.info(f"Start sending letter for notification id: {notification_id}")
+    notification = notifications_dao.get_notification_by_id(notification_id, _raise=True)
+
+    if notification.template.is_precompiled_letter:
+        address_lines = PostalAddress(notification.to, allow_international_letters=True).normalised_lines
+    else:
+        address_lines = PostalAddress.from_personalisation(
+            InsensitiveDict(notification.personalisation),
+            allow_international_letters=True,
+        ).normalised_lines
+
+    try:
+        file_bytes = find_letter_pdf_in_s3(notification).get()["Body"].read()
+    except (BotoClientError, LetterPDFNotFound) as e:
+        current_app.logger.exception(f"Error getting letter from bucket for notification: {notification_id}", str(e))
+        return
+
+    try:
+        dvla_client.send_letter(
+            notification_id=str(notification.id),
+            address=address_lines,
+            postage=notification.postage,
+            service_id=str(notification.service_id),
+            organisation_id=str(notification.service.organisation_id),
+            pdf_file=file_bytes,
+        )
+        update_letter_to_sending(notification)
+    except DvlaRetryableException as e:
+        if isinstance(e, DvlaThrottlingException):
+            current_app.logger.warning(f"RETRY: Letter notification {notification_id} was rate limited by DVLA")
+        else:
+            current_app.logger.exception(f"RETRY: Letter notification {notification_id} failed")
+
+        self.retry()
+    except DvlaNonRetryableException as e:
+        if isinstance(e, DvlaDuplicatePrintRequestException):
+            current_app.logger.warning(f"Duplicate deliver_letter task called for notification {notification_id}")
+            return
+
+        update_notification_status_by_id(notification_id, NOTIFICATION_TECHNICAL_FAILURE)
+        raise NotificationTechnicalFailureException(f"Error when sending letter notification {notification_id}") from e
+    except self.MaxRetriesExceededError as e:
+        update_notification_status_by_id(notification_id, NOTIFICATION_TECHNICAL_FAILURE)
+        raise NotificationTechnicalFailureException(
+            f"RETRY FAILED: Max retries reached. The task deliver_letter failed for notification {notification_id}. "
+            "Notification has been updated to technical-failure"
+        ) from e
```

Comment:
I believe this is in the wrong place - it needs to wrap the `self.retry` as that is what will throw this exception (see `deliver_email` and `deliver_sms`)

Reply:
Yep, you're right - I've moved it and added the change as a new commit (instead of rebasing) to make it easier to review the difference